File: AkitaModel/Results/VisitBalance.cs
using System.Collections.Generic;
using System.Linq;

namespace Skyware.Lis.AkitaModel.Results;

/// <summary>
/// Represents a Visit view with structure for bill reporting.
/// </summary>
public class VisitBalance : VisitBase
{

    /// <summary>
    /// List of sales.
    /// </summary>
    public IList<VisitSale> Sales { get; set; }

    /// <summary>
    /// Balance of the visit.
    /// </summary>
    public decimal Balance => Sales?.Sum(s => s.Balance) ?? 0m;

}

File: AkitaMotelTS/src/Results/VisitBalance.ts
/**
 * Skyware.Lis.AkitaModel.Results
 */

import { VisitBase } from '../VisitBase';
import { VisitSale } from './VisitSale';

/**
 * Represents a Visit view with structure for bill reporting.
 */
export class VisitBalance extends VisitBase {
  /**
   * List of sales.
   */
  sales?: VisitSale[];

  /**
   * Balance of the visit.
   */
  get balance(): number {
    return this.sales?.reduce((sum, s) => sum + s.balance, 0) ?? 0;
  }
}
